From 8c1189c39ba3f28b9f428d96469747fa794b4273 Mon Sep 17 00:00:00 2001
From: Jason Hung <jhung@globalscaletechnologies.com>
Date: Wed, 25 Sep 2024 13:49:12 +0800
Subject: [PATCH 5/5] dts: gti-mt510ss-evk: add external amplifier control for
 Speaker

add external amplifier control for Speaker
 - add amp-gpio for turn on / off ext-amp
 - change audio routing
---
 arch/arm64/boot/dts/mediatek/gti-mt510ss-evk.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/gti-mt510ss-evk.dts b/arch/arm64/boot/dts/mediatek/gti-mt510ss-evk.dts
index 3a121a69d0b4..4290d97ead26 100644
--- a/arch/arm64/boot/dts/mediatek/gti-mt510ss-evk.dts
+++ b/arch/arm64/boot/dts/mediatek/gti-mt510ss-evk.dts
@@ -1071,6 +1071,7 @@ &keypad {
 &mt6359codec {
 	mediatek,mic-type-0 = <1>; /* ACC */
 	mediatek,mic-type-1 = <3>; /* DCC */
+	amp-gpio = <&pio 13 GPIO_ACTIVE_HIGH>;
 };
 
 &mt6359_vbbck_ldo_reg {
@@ -1361,6 +1362,8 @@ &sound {
 	audio-routing =
 		"Headphone", "Headphone L",
 		"Headphone", "Headphone R",
+		"Speaker", "Speaker L",
+		"Speaker", "Speaker R",
 		"AIN1", "Headset Mic",
 		"DMIC_INPUT", "AP DMIC",
 		"AP DMIC", "AUDGLB",
-- 
2.34.1

